tests/test_blob_repairer.py

```diff
@@ -408,6 +416,74 @@ def test_check_index_non_valid(_mocked_exists,
     assert blob_repairer.stat.index_malformed_headers == 0
 
 
+@pytest.mark.parametrize('callee', [
+    BlobRepairer.recover_index,
+    BlobRepairer.recover_blob,
+    BlobRepairer.copy_valid_records,
+])
+@mock.patch('eblob_kit.is_destination_writable', return_value=True)
+@mock.patch(OPEN_TO_PATCH, new_callable=mock.mock_open)
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
+def test_fix_destination_writable(_mock_exist,
+                                  mocked_data_file,
+                                  _mocked_index_file,
+                                  _mocked_open,
+                                  _mocked_is_writable,
+                                  mocker,
+                                  callee):
+    """Check for exception if destination not writable.
+
+    Checks for `copy_valid_records`, `recover_index` and `recover_blob`.
+
+    """
+    blob_repairer = BlobRepairer('.')
+
+    assert blob_repairer.valid
+    assert not blob_repairer.stat.index_order_error
+
+    if isinstance(callee, types.FunctionType):
+        # BlobRepairer.recover_index static method
+        callee(blob_repairer._blob.data, '.')
+    else:
+        callee(blob_repairer, '.')
+
+
+@pytest.mark.parametrize('callee', [
+    BlobRepairer.recover_index,
+    BlobRepairer.recover_blob,
+    BlobRepairer.copy_valid_records,
+])
+@mock.patch('eblob_kit.is_destination_writable', return_value=False)
+@mock.patch(OPEN_TO_PATCH, new_callable=mock.mock_open)
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
```

Comment:
Why is it mocked?

Reply:
It is used somewhere in `BlobRepairer` on construction.

Reply:
Could you please find this place? Because `os.path.exists` is called in several places and some of them looks like shouldn't be unconditionally True.

Reply:
Please refer to `Blob.__init__`:
https://github.com/interiorem/eblob_kit/blob/4a07d68582b52cf5e8c8c3e4331a71d5aa7423e6/eblob_kit.py#L524

Reply:
And seems that it doesn't resonate with code being tested.